src/Cli/dotnet/commands/dotnet-workload/WorkloadInfoHelper.cs

```diff
@@ -31,12 +31,14 @@ internal class WorkloadInfoHelper : IWorkloadInfoHelper
             string userProfileDir = null,
             IWorkloadResolver workloadResolver = null)
         {
-            DotnetPath = dotnetDir ?? Path.GetDirectoryName(Environment.ProcessPath);
             ReleaseVersion currentSdkReleaseVersion = new(currentSdkVersion ?? Product.Version);
             _currentSdkFeatureBand = new SdkFeatureBand(currentSdkReleaseVersion);
 
             _targetSdkVersion = targetSdkVersion;
             userProfileDir ??= CliFolderPathCalculator.DotnetUserProfileFolderPath;
+            DotnetPath = dotnetDir ?? (WorkloadFileBasedInstall.IsUserLocal(Path.GetDirectoryName(Environment.ProcessPath), _currentSdkFeatureBand.ToString()) ?
```

Comment:
Rather than setting the DotnetPath to the user local workload folder, I think we should have a separate property for the workload root.  In some contexts we still want the actual dotnet root folder, for example when creating the `SdkDirectoryWorkloadManifestProvider` and workload resolver.  These constructors have separate arguments to specify the user profile directory.